src/simulate.jl

```diff
@@ -37,7 +37,98 @@ function simulate!(
     m
 end
 
-function simulate!(m::LinearMixedModel{T}; β = coef(m), σ = m.σ, θ = T[]) where {T}
+function simulate!(
+    rng::AbstractRNG,
+    m::GeneralizedLinearMixedModel{T};
+    β = coef(m),
+    σ = m.σ,
+    θ = T[],
+) where {T}
+    length(β) == length(fixef(m)) ||
+        length(β) == length(coef(m)) ||
+            throw(ArgumentError("You must specify all (non-singular) βs"))
+
+    dispersion_parameter(m) || isnan(σ) ||
+        throw(ArgumentError("You must not specify a dispersion parameter for model families without a dispersion parameter"))
+
+    β = convert(Vector{T},β)
+    σ = T(σ)
+    θ = convert(Vector{T},θ)
+
+    d = m.resp.d
+
+    if length(β) ≠ length(coef(m))
+        padding = length(coef(m)) - length(β)
+        for ii in 1:padding
+            push!(β, -0.0)
+        end
+    end
+
+    fast = (length(m.θ) == length(m.optsum.final))
+    setpar! = fast ? setθ! : setβθ!
+    params = fast ? θ : vcat(β, θ)
+    setpar!(m, params)
+
+    lm = m.LMM
+
+    # note that these m.resp.y and m.LMM.y will later be sychronized in (re)fit!()
+    # but for now we use them as distinct scratch buffers to avoid allocations
+
+    # the noise term is actually in the GLM and not the LMM part so no noise
+    # at the LMM level
+    η = fill!(m.LMM.y, zero(T))
+    y = m.resp.y
+
+    # assemble the linear predictor
+    @inbounds for trm in m.reterms             # add the unscaled random effects
+        unscaledre!(rng, η, trm)
+    end
+
+    # scale by lm.σ and add fixed-effects contribution
+    BLAS.gemv!('N', one(T), lm.X, β, lm.σ, η)
+
+    # from η to μ
+    GLM.updateμ!(m.resp, η)
+
+    # convert to the distribution / add in noise
+    @inbounds for (idx, val) in enumerate(m.resp.mu)
+        n = isempty(m.wt) ? 1 : m.wt[idx]
+        y[idx] = _rand(rng, d, val, σ, n)
+    end
+
+    m
+end
+
+"""
+    _rand(rng::AbstractRNG, d::Distribution, location, scale=NaN, n=1)
+
+A convenience function taking a draw from a distrbution.
+
+Note that `d` is specified as an existing distribution, such as
+from the `GlmResp.d` field. This isn't vectorized nicely because
+for distributions where the scale/dispersion is dependent on the
+location (e.g. Bernoulli, Binomial, Poisson), it's not really
+possible to avoid creating multiple `Distribution` objects.
+
+Note that `n` is the `n` parameter for the Binomial distribution,
+*not* the number of draws from the RNG. It is then used to change the
+random draw (an integer in [0, n]) into a probability (a float in [0,1]).
+"""
+function _rand(rng::AbstractRNG, d::Distribution, location, scale=NaN, n=1)
```

Comment:
Maybe use `missing` instead of `NaN` to indicate that `scale` is not defined?

Reply:
Right now, `GeneralizedLinearMixedModel.sigma` yields `sdest(::GeneralizedLinearMixedModel)` which returns `NaN` for models without a dispersion parameter. I have no objections to changing this -- it's in a strict sense breaking, but

1. Users really shouldn't be invoking `sdest`, `varest` or `dispersion` on models without a dispersion parameter. 
2. We already warn that the behavior for models with dispersion parameter isn't correct.
3. I would view fixing this behavior as a bugfix, which are allowed to be breaking to fix incorrect behavior.

My deep dive into the GLMM code this last week has highlighted to me that these definitions are inconsistent. Currently, For models  without a dispersion parameter:
1. `dispersion` is defined to be 1. This follows the behavior of `GLM.jl` for Bernoulli, Binomial and Poisson.
2. `varest` is `NaN`. This is a MixedModels function, not defined in `StatsBase` nor `GLM`.
3. `sdest` is `NaN`. This is a MixedModels function, not defined in `StatsBase` nor `GLM`.

I think it makes sense to have `varest` and `sdest` return `missing` for Bernoulli, Binomial and Poisson and document that this differs from the behavior of `dispersion` (much the same way that `coef` and `fixef` do similar but not identical things). Thoughts?

Reply:
I've now implemented this suggestion.

Reply:
Thanks for implementing this and sorry that I haven't responded earlier.  Living in a swing state at this point in the election is nerve wracking and my productivity has plummeted.